test/parameter_test.py

```diff
@@ -1329,3 +1335,29 @@ def test_exists(self, path_parameter):
                 luigi.build([path_parameter["cls"]()], local_scheduler=True)
         else:
             assert luigi.build([path_parameter["cls"]()], local_scheduler=True)
+
+
+if Annotated is None:
+    PEP593Task = luigi.Task
+else:
+    class PEP593Task(luigi.Task):
+        param: Annotated[int, luigi.IntParameter()]
```

Comment:
I think it would be nice to check cases like `param: Annotated[int, luigi.BoolParameter()]` like suggested in another comment